data_steward/analytics/cdr_ops/cross_ct_rt/dst_251_addtional_QCs_for_the_Controlled_Tier_part1.py

```diff
@@ -0,0 +1,970 @@
+# ---
+
+# # DST-251 Add more QA/QCs for the Controlled Tier data
+#
+# We want to mirror the QCs in here   to add more checks to the Controlled Tier (CT).
+#
+# The privacy checks against the CT are done. But we need more QCs that check against the OMOP rules AND against the Program constraints. 
+#
+# Create a notebook for the following checks AND commit them here. If you are just starting working WITH curation, follow the instructions here  .
+#
+# The checks are:
+#
+# 1 No dates earlier than birth dates
+#
+# 2 No dates earlier than 1980 in any table, except for the observation
+#
+# 3 No dates after death:done
+#
+# 4 No WITHdrawn participants: need pdr access
+#
+# 5 No data after participant's suspension: need pdr access
+#
+# 6 All participants have basics
+#
+# 7 All participants WITH EHR data have said yes to EHR consents
+#
+# 8 All participants WITH Fitbit have said yes to primary consents
+#
+# 9 All primary keys are in _ext
+#
+# 10 No duplicated primary keys
+#
+# 11 OMOP tables should have standard concept ids
+#
+# 12 observation concept ids (4013886, 4135376, 4271761) that have dates equal to birth dates should be set to CDR cutoff date
+#
+# 13 all other observation concept ids WITH dates similar to birth dates other than the 3 above should be removed
+#
+# 14 All the descendants of ancestor_concept_id IN (4054924, 141771) -- motor vehicle accidents should be dropped in condition_occurrence table
+
+# + tags=["parameters"]
+# Parameters
+project_id = ""
+rt_dataset = ""
+ct_dataset = ""
+earliest_ehr_date=""
+cut_off_date=""
+
+# +
+import pandas as pd
+from analytics.cdr_ops.notebook_utils import execute
+from common import JINJA_ENV
+from utils import auth
+from utils.bq import get_client
+
+client = get_client(project_id)
+
+pd.options.display.max_rows = 120
+# -
+
+# summary will have a summary in the end
+df = pd.DataFrame(columns = ['query', 'result']) 
+
+# # Query1: No dates before birth_date
+#
+
+# +
+# step1 , to get the tables AND columns that have person_id, size >1 AND DATE columns AND save to a data frame
+query = JINJA_ENV.from_string("""
+
+WITH
+    table1 AS (
+    SELECT
+      table_name,
+      column_name
+    FROM
+      `{{project_id}}.{{ct_dataset}}.INFORMATION_SCHEMA.COLUMNS`
+    WHERE 
+      column_name='person_id' ),
+    table2 AS (
+    SELECT
+      table_id AS table_name,
+      row_count
+    FROM
+      `{{project_id}}.{{ct_dataset}}.__TABLES__`
+    WHERE 
+      row_count>1)
+      
+  SELECT
+    table_name,
+    column_name
+  FROM
+    `{{project_id}}.{{ct_dataset}}.INFORMATION_SCHEMA.COLUMNS` c
+  WHERE 
+    table_name IN (
+    SELECT
+      DISTINCT table_name
+    FROM
+      table2
+    WHERE 
+      table_name IN (
+      SELECT
+        DISTINCT table_name
+      FROM
+        table1))
+    AND c.data_type IN ('DATE')
+    AND NOT REGEXP_CONTAINS(column_name, r'(?i)(_PAR)') 
+""")
+
+q = query.render(project_id=project_id, rt_dataset=rt_dataset, ct_dataset=ct_dataset)
```

Comment:
rt_dataset is not being used in this query.

Reply:
yes, you may see several places like this, since such 'unused' variable did not have any impact when running the cell, I just went on without concerning. Again, we sure can clean this up.